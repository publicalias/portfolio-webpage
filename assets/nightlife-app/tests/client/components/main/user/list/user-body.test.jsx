"use strict";

//local imports

const UserBody = require("../../../../../../scripts/client/components/main/user/list/user-body");

const { testWrapper } = require("../../../../test-helpers");

//global imports

const { initTestSnapshot, withDataList } = require("redux/tests/client-tests");
const { reactTests } = require("redux/tests/react-tests");

//utilities

const { testShallow } = testWrapper(UserBody);

//setup

beforeAll(reactTests.setup);
beforeEach(reactTests.inject(UserBody));

//user body

describe("UserBody", () => {

  const testSnapshot = withDataList(initTestSnapshot(testShallow), [null, { handleScroll: jest.fn() }]);

  it("should match snapshot (default)", () => testSnapshot());

  it("should match snapshot (list)", () => testSnapshot({ users: { data: [{}] } }));

});
