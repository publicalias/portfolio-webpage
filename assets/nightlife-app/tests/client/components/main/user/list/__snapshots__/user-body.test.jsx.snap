// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBody should match snapshot (default) 1`] = `
<MetaListBody
  local={
    Object {
      "handleScroll": [MockFunction],
      "header": <div
        className="c-list-body__header"
      >
        <h1>
          Users
        </h1>
      </div>,
      "list": Array [],
      "placeholder": "No Users",
    }
  }
/>
`;

exports[`UserBody should match snapshot (list) 1`] = `
<MetaListBody
  local={
    Object {
      "handleScroll": [MockFunction],
      "header": <div
        className="c-list-body__header"
      >
        <h1>
          Users
        </h1>
      </div>,
      "list": Array [
        <UserItem
          local={
            Object {
              "user": Object {
                "data": Object {
                  "address": "",
                  "app": "nightlife-app",
                  "avatar": "",
                  "blocks": Array [],
                  "distance": 0,
                  "favorites": Array [],
                  "friends": Array [],
                  "location": null,
                },
                "id": "",
                "name": "",
              },
            }
          }
        />,
      ],
      "placeholder": "No Users",
    }
  }
/>
`;
